tests/hil/scripts/pytest-hil/board.py

```diff
@@ -1,46 +1,84 @@
 from abc import ABC, abstractmethod
+from contextlib import asynccontextmanager
 import serial
 from time import time, sleep
 import re
 
+import trio
+from trio_serial import SerialStream
+
 class Board(ABC):
-    def __init__(self, port, baud, wifi_ssid, wifi_psk, fw_image, serial_number):
+    def __init__(self, port, baudrate, wifi_ssid, wifi_psk, fw_image, serial_number):
         if serial_number:
             self.serial_number = serial_number
         self.port = port
+        self.baudrate = baudrate
+        self.fw_image = fw_image
+        self.wifi_ssid = wifi_ssid
+        self.wifi_psk = wifi_psk
+
+        self.received: bytes = b''
+
+        self.serial: SerialStream | None = None
+
+    @asynccontextmanager
+    async def started(self):
+        async with SerialStream(port=self.port,
+                                baudrate=self.baudrate) as serial:
+            self.serial = serial
 
-        if fw_image:
-            self.program(fw_image)
+            if self.fw_image:
+                self.program(self.fw_image)
```

Comment:
I wonder if the issue with the ESP-IDF tests is that we are programming the board _after_ opening the SerialStream, since the Espressif devkits are programmed over serial.

Reply:
This indeed made it work. Great catch!